src/display/text_layer.js

```diff
@@ -540,59 +557,57 @@ var renderTextLayer = (function renderTextLayerClosure() {
-          var div = this._textDivs[i];
-          var transform;
-          var width = div.dataset.originalWidth;
-          if (div.dataset.canvasWidth !== undefined && width > 0) {
-            // Dataset values are of type string.
-            var textScale = div.dataset.canvasWidth / width;
+
+      for (var i = 0, ii = this._textDivs.length; i < ii; i++) {
+        var div = this._textDivs[i];
+        var divProperties = this._textDivProperties.get(div);
+
+        if (expandDivs) {
+          var transform = '';
+          var width = divProperties.originalWidth;
+          var textScale = 1;
+
+          if (divProperties.canvasWidth !== 0 && width > 0) {
+            textScale = divProperties.canvasWidth / width;
             transform = 'scaleX(' + textScale + ')';
-          } else {
-            transform = '';
           }
-          var rotation = div.dataset.angle;
-          if (rotation) {
+          var rotation = divProperties.angle;
+          if (rotation !== 0) {
             transform = 'rotate(' + rotation + 'deg) ' + transform;
           }
-          if (div.dataset.paddingLeft) {
+          if (divProperties.paddingLeft !== 0) {
             div.style.paddingLeft =
-              (div.dataset.paddingLeft / textScale) + 'px';
+              (divProperties.paddingLeft / textScale) + 'px';
             transform += ' translateX(' +
-              (-div.dataset.paddingLeft / textScale) + 'px)';
+              (-divProperties.paddingLeft / textScale) + 'px)';
           }
-          if (div.dataset.paddingTop) {
-            div.style.paddingTop = div.dataset.paddingTop + 'px';
-            transform += ' translateY(' + (-div.dataset.paddingTop) + 'px)';
+          if (divProperties.paddingTop !== 0) {
+            div.style.paddingTop = divProperties.paddingTop + 'px';
+            transform += ' translateY(' + (-divProperties.paddingTop) + 'px)';
           }
-          if (div.dataset.paddingRight) {
+          if (divProperties.paddingRight !== 0) {
             div.style.paddingRight =
-            div.dataset.paddingRight / textScale + 'px';
+              divProperties.paddingRight / textScale + 'px';
           }
-          if (div.dataset.paddingBottom) {
-            div.style.paddingBottom = div.dataset.paddingBottom + 'px';
+          if (divProperties.paddingBottom !== 0) {
+            div.style.paddingBottom = divProperties.paddingBottom + 'px';
           }
-          if (transform) {
-            CustomStyle.setProp('transform' , div, transform);
+          if (transform !== '') {
+            CustomStyle.setProp('transform', div, transform);
           }
-        }
-      } else {
-        for (i = 0, ii = this._textDivs.length; i < ii; i++) {
-          div = this._textDivs[i];
+        } else {
           div.style.padding = 0;
-          transform = div.dataset.originalTransform || '';
+          transform = divProperties.originalTransform;
```

Comment:
Isn't `transform` only defined in the `else` block thanks to [variable hoisting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var#var_hoisting), which is something that we generally try to avoid. You could probably just do `CustomStyle.setProp('transform', div, divProperties.originalTransform);` instead.